File: src/dataset/c/triangle_Exercism_vlzware/openAI/ChatGPT4_triangle_v1.c
#include "triangle.h"

static inline int is_valid_triangle(triangle_t t)
{
    return (t.a > 0 && t.b > 0 && t.c > 0) &&
           (t.a < t.b + t.c) &&
           (t.b < t.a + t.c) &&
           (t.c < t.a + t.b);
}

int is_equilateral(triangle_t t)
{
    return (t.a == t.b && t.a == t.c) && is_valid_triangle(t);
}

int is_isosceles(triangle_t t)
{
    return ((t.a == t.b || t.a == t.c || t.b == t.c) && is_valid_triangle(t));
}

int is_scalene(triangle_t t)
{
    return ((t.a != t.b && t.a != t.c && t.b != t.c) && is_valid_triangle(t));
}

File: src/dataset/typescript/triangle_Exercism-typescript-alemarr/openAI/ChatGPT4_triangle_v1.ts
type TriangleKind = "equilateral" | "isosceles" | "scalene";

export class Triangle {
  private readonly isValid: boolean;

  constructor(private a: number, private b: number, private c: number) {
    this.isValid = this.a > 0 && this.b > 0 && this.c > 0 &&
                   this.a + this.b >= this.c &&
                   this.b + this.c >= this.a &&
                   this.c + this.a >= this.b;
  }

  get isEquilateral(): boolean {
    return this.isValid && this.a === this.b && this.b === this.c;
  }

  get isIsosceles(): boolean {
    return this.isValid && (this.isEquilateral || this.a === this.b || this.b === this.c || this.c === this.a);
  }

  get isScalene(): boolean {
    return this.isValid && this.a !== this.b && this.b !== this.c && this.c !== this.a;
  }
}